File: src/main/java/ac/dnd/server/common/propeties/CorsProperties.java
package ac.dnd.server.common.propeties;

import java.util.ArrayList;
import java.util.List;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import lombok.Getter;

@Component
@ConfigurationProperties(prefix = "cors")
@Getter
public class CorsProperties {
	private List<String> allowedOriginsPatterns = new ArrayList<>();
	private List<String> allowedHeaders = new ArrayList<>();
	private List<String> exposedHeaders = new ArrayList<>();

	public void setAllowedOriginsPatterns(final List<String> allowedOriginsPatterns) {
		this.allowedOriginsPatterns = allowedOriginsPatterns;
	}

	public void setAllowedHeaders(final List<String> allowedHeaders) {
		this.allowedHeaders = allowedHeaders;
	}

	public void setExposedHeaders(final List<String> exposedHeaders) {
		this.exposedHeaders = exposedHeaders;
	}
}

File: src/main/kotlin/ac/dnd/server/common/propeties/CorsProperties.kt
package ac.dnd.server.common.propeties

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "cors")
class CorsProperties(
    val allowedOriginsPatterns: List<String> = emptyList(),
    val allowedHeaders: List<String> = emptyList(),
    val exposedHeaders: List<String> = emptyList(),
    val allowedMethods: List<String> = emptyList(),
    val allowCredentials: Boolean = false,
    val maxAge: Long = 3600L
)